packages/scandipwa/src/component/ProductBundleItems/ProductBundleItems.container.js

```diff
@@ -73,42 +78,94 @@ export class ProductBundleItemsContainer extends ProductCustomizableOptionsConta
-        let price = 0;
-        let priceExclTax = 0;
+    optionPriceMap = {
+        [PRICE_TYPE_DYNAMIC]: this.getDynamicOptionPrice.bind(this),
+        [PRICE_TYPE_FIXED]: this.getFixedOptionPrice.bind(this),
+        [PRICE_TYPE_PERCENT]: this.getPercentOptionPrice.bind(this)
+    };
+
+    getDynamicOptionPrice(product, quantity) {
+        const {
+            price_range: {
+                minimum_price: {
+                    regular_price: {
+                        value: itemPrice = 0
+                    } = {},
+                    final_price: {
+                        value: finalItemPrice = 0
+                    } = {},
+                    final_price_excl_tax: {
+                        value: finalItemPriceExclTax = 0
+                    } = {}
+                } = {}
+            } = {}
+        } = product;
+
+        const optionInitialPrice = itemPrice * quantity;
+        const optionPrice = finalItemPrice * quantity;
+        const optionPriceExclTax = finalItemPriceExclTax * quantity;
+        return { optionPrice, optionPriceExclTax, optionInitialPrice };
+    }
+
+    getPercentOptionPrice(product, quantity, optionPricePercent) {
+        const {
+            price_range: {
+                minimum_price: {
+                    default_price: { value: defaultPrice } = {},
+                    default_final_price: { value: defaultFinalPrice } = {},
+                    default_final_price_excl_tax: { value: defaultFinalPriceExclTax } = {}
+                }
+            }
+        } = this.props;
+
+        const optionPriceMultiplier = optionPricePercent / ONE_HUNDRED_PERCENT;
+
+        const optionInitialPrice = defaultPrice * optionPriceMultiplier * quantity;
+        const optionPrice = (defaultFinalPrice * optionPriceMultiplier * quantity);
+        const optionPriceExclTax = (defaultFinalPriceExclTax * optionPriceMultiplier * quantity);
+        return { optionPrice, optionPriceExclTax, optionInitialPrice };
+    }
+
+    getFixedOptionPrice(product, quantity, optionPriceFixed) {
+        const optionInitialPrice = optionPriceFixed * quantity;
+        const optionPrice = optionPriceFixed * quantity;
+        const optionPriceExclTax = optionPriceFixed * quantity;
+        return { optionPrice, optionPriceExclTax, optionInitialPrice };
+    }
 
-        selectedValues.forEach(({ id, quantity, value }) => {
-            if (option_id === id) {
-                const { options } = item;
+    getOptionPrice(item, selectedValues, isDynamicPrice) {
+        const { option_id, options } = item;
 
-                options.forEach(({ id: optionId, product }) => {
+        return selectedValues
+            .filter(({ id }) => id === option_id)
+            .reduce((acc, { quantity, value }) => {
+                const { price, priceExclTax, initialPrice } = acc;
+
+                const {
+                    optionPrice, optionPriceExclTax, optionInitialPrice
+                } = options.reduce((acc, {
```

Comment:
Also, I believe `reduce` here is not necessary because it skips all items that don't match some condition. Prefer using `filter` or `find` here, it's more straightforward.